core/src/main/scala/filodb.core/memstore/ratelimit/CardinalityTracker.scala

```diff
@@ -61,28 +61,33 @@ class CardinalityTracker(ref: DatasetRef,
     // modifyCount and decrementCount methods are protected this way
 
     require(shardKey.length == shardKeyLen, "full shard key is needed")
-    require(totalDelta == 1 && activeDelta == 0 ||   // new ts but inactive
-            totalDelta == 1 && activeDelta == 1 ||   // new ts and active
-            totalDelta == 0 && activeDelta == 1 ||   // // existing inactive ts that became active
-            totalDelta == 0 && activeDelta == -1,    // existing active ts that became inactive
-            "invalid values for totalDelta / activeDelta")
+    require(totalDelta == 1 && activeDelta == 0 || // new ts but inactive
+      totalDelta == 1 && activeDelta == 1 || // new ts and active
+          totalDelta == 0 && activeDelta == 1 ||   // // existing inactive ts that became active
+      totalDelta == 0 && activeDelta == -1, // existing active ts that became inactive
+      "invalid values for totalDelta / activeDelta")
 
     val toStore = ArrayBuffer[(Seq[String], CardinalityRecord)]()
     // first make sure there is no breach for any prefix
     (0 to shardKey.length).foreach { i =>
       val prefix = shardKey.take(i)
       val old = store.getOrZero(prefix, CardinalityRecord(shard, prefix, 0, 0, 0, defaultChildrenQuota(i)))
+      val newActive = Math.max(old.activeTsCount + activeDelta, 0)
+      if (old.activeTsCount + activeDelta < 0) {
+        logger.error(s"For some reason, activeTs count for prefix $prefix cannot be reduced below zero",
+          new RuntimeException())
```

Comment:
I get that you're just logging a stack trace here, but I wonder if there will be any performance implications if it turns out this error happens quite often.

Reply:
This should be temporary until we fix this.

Reply:
removed stack trace and added other relevant information